src/v/cluster/tests/partition_allocator_tests.cc

```diff
@@ -533,9 +566,9 @@ FIXTURE_TEST(rack_aware_assignment_1, partition_allocator_fixture) {
         nodes.insert(node_id);
     }
     BOOST_REQUIRE(nodes.size() == 3);
-    BOOST_REQUIRE(nodes.contains(model::node_id(0))); // rack-a
-    BOOST_REQUIRE(nodes.contains(model::node_id(2))); // rack-b
-    BOOST_REQUIRE(nodes.contains(model::node_id(4))); // rack-c
+    BOOST_REQUIRE(nodes.contains(model::node_id(0)) || nodes.contains(model::node_id(1))); // rack-a
+    BOOST_REQUIRE(nodes.contains(model::node_id(2)) || nodes.contains(model::node_id(3))); // rack-b
+    BOOST_REQUIRE(nodes.contains(model::node_id(4)) || nodes.contains(model::node_id(5))); // rack-c
```

Comment:
This existing test relied on the exact allocator behavior to check the result: in fact either node could be chosen as they meet the rack-based constraints, so loosen the check to include either node.